### PR TITLE
Change the Woo logos in the mobile banner and modal in admin

### PR DESCRIPTION
**Pull request by Konamiman**
### Changes proposed in this Pull Request:

This pull request is a follow-up of https://github.com/woocommerce/woocommerce/pull/53308, it changes the Woo logo in two places:

1. The "Download the WooCommerce App" banner that appears in the WooCommerce dashboard in Android devices:

image

2. The "Manage orders and track sales in real-time with the free mobile app" modal window that appears when clicking "Get the free WooCommerce mobile app" link in the "What to do next" box in the WooCommerce dashboard (the mobile devices image and its background pattern are changed too):

image

image

### How to test the changes in this Pull Request:

1. Activate the new branding as instructed in https://github.com/woocommerce/woocommerce/pull/53308

2. Open the WooCommerce dahsboard (WooCommerce - Home) from an Android device. Alternatively, you can simulate such a device using the developer tools of your browser, here's how in Chrome:

* Open the developer tools (F12 key)
* Click the "Toggle device toolbar" icon:

image

* Click on the device selector and select "Edit"
* Create a new device with the word "Android" in the user agent string

3. Verify that the logo next to the "Download the WooCommerce App" text is the new one.

Note: if you have seen and dismissed this banner in the past it won't appear. To restore the logo you can run this in the command line (replace the user id accordingly):

```
wp db query "delete from wp_usermeta where meta_key='woocommerce_admin_android_app_banner_dismissed' and user_id=<user id>"
```

4. Go back to normal display (click the "Toggle device toolbar" icon again) and click the "Get the free WooCommerce mobile app" link in the "What to do next" box. Verify that you see the new logo in the top left corner of the modal window that opens.

5. Deactivate the new branding, repeat the steps, and verify that you see the old logos again.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>

### CODE REVIEW

**Comment by barryhughes** [comment]
A little unsure if this is a 'real' problem or not, but I notice the live branch didn't build, and these errors referencing the new .scss files were present:

```
client/admin build:project:bundle: ERROR in ./client/homescreen/mobile-app-modal/index.tsx:47:10
client/admin build:project:bundle: TS2307: Cannot find module './style-new.scss' or its corresponding type declarations.
client/admin build:project:bundle:     45 |
client/admin build:project:bundle:     46 | 	if ( isNewBranding() ) {
client/admin build:project:bundle:   > 47 | 		import('./style-new.scss');
client/admin build:project:bundle:        | 		       ^^^^^^^^^^^^^^^^^^
client/admin build:project:bundle:     48 | 	}
client/admin build:project:bundle:     49 | 	else {
client/admin build:project:bundle:     50 | 		import('./style-old.scss');
client/admin build:project:bundle: ERROR in ./client/homescreen/mobile-app-modal/index.tsx:50:10
client/admin build:project:bundle: TS2307: Cannot find module './style-old.scss' or its corresponding type declarations.
client/admin build:project:bundle:     48 | 	}
client/admin build:project:bundle:     49 | 	else {
client/admin build:project:bundle:   > 50 | 		import('./style-old.scss');
client/admin build:project:bundle:        | 		       ^^^^^^^^^^^^^^^^^^
```

**Comment by barryhughes** [comment]
Hmm, running `pnpm build` locally for this branch also fails, though I'm less clear how this error relates to this PR:

```
plugins/woocommerce/client/legacy build:project:assets: Running "sass:compile" (sass) task
plugins/woocommerce/client/legacy build:project:assets: Fatal error: Error: Can't find stylesheet to import.
plugins/woocommerce/client/legacy build:project:assets:   ╷
plugins/woocommerce/client/legacy build:project:assets: 9 │ @import "@automattic/color-studio/dist/color-variables";
plugins/woocommerce/client/legacy build:project:assets:   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
plugins/woocommerce/client/legacy build:project:assets:   ╵
plugins/woocommerce/client/legacy build:project:assets:   css/admin/product-editor.scss 9:9  @import
plugins/woocommerce/client/legacy build:project:assets:   css/admin.scss 8623:9              root stylesheet
plugins/woocommerce/client/legacy build:project:assets: ❌ [build:project:assets] exited with exit code 1.
plugins/woocommerce/client/legacy build:project:assets: ❌ 1 script failed.
plugins/woocommerce/client/legacy build:project:assets: Failed

```

**Comment by Konamiman** [comment]
> A little unsure if this is a 'real' problem or not, but I notice the live branch didn't build, and these errors referencing the new .scss files were present

That's Node freaking out because we are doing a dynamic import of a stylesheet (static imports, like the one at the top of the file, seem to be fine); but it works as expected nonetheless. I wouldn't worry about this since it's a temporary change, once the rebranding is permanent and the feature flag is removed these imports will go away too.

**Edit:** I have added a `declaration.d.ts` file that makes the build finish without errors.

> though I'm less clear how this error relates to this PR:

I think this has been discussed in a different PR, tldr: it's related to the rebranding but indeed not to this specific pull request.

I have pushed an additional commit to change the mobile devices image (a bit difficult to notice, but there are color changes in a few of the links) and its background pattern, can you please verify those as well?

**Comment by barryhughes** [comment]
For some reason `pnpm build` still fails for me, with this branch, but in any case the live branch task completed so I can verify that way :+1:

File: plugins/woocommerce/client/admin/client/homescreen/mobile-app-modal/style-new.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
...375 C 30.15625 28.984375 32.292969 26.894531 32.308594 24.082031 C 32.308594 21.257812 30.15625 19.167969 27.359375 19.167969 Z M 27.359375 25.964844 C 26.304688 25.964844 25.578125 25.175781 25.578125 24.082031 C 25.578125 22.988281 26.304688 22.183594 27.359375 22.183594 C 28.414062 22.183594 29.140625 22.988281 29.140625 24.082031 C 29.140625 25.175781 28.429688 25.964844 27.359375 25.964844 Z M 27.359375 25.964844 "/></g></g><g clip-path="url(%23clip6)" clip-rule="nonzero"><g clip-path="url(%23clip7)" clip-rule="nonzero"><path style=" stroke:none;fill-rule:nonzero;fill:rgb(100%,100%,100%);fill-opacity:1;" d="M 37.957031 19.167969 C 35.160156 19.167969 33.007812 21.257812 33.007812 24.082031 C 33.007812 26.910156 35.160156 28.984375 37.957031 28.984375 C 40.753906 28.984375 42.90625 26.894531 42.90625 24.082031 C 42.90625 21.269531 40.753906 19.167969 37.957031 19.167969 Z M 37.957031 25.964844 C 36.886719 25.964844 36.1875 25.175781 36.1875 24.082031 C 36.1875 22.988281 36.902344 22.183594 37.957031 22.183594 C 39.011719 22.183594 39.738281 22.988281 39.738281 24.082031 C 39.738281 25.175781 39.011719 25.964844 37.957031 25.964844 Z M 37.957031 25.964844 "/></g></g></g></svg>')
```
Comment (barryhughes) [inline plugins/woocommerce/client/admin/client/homescreen/mobile-app-modal/style-new.scss]: Noting there are [linting failures](https://github.com/woocommerce/woocommerce/actions/runs/12355468424/job/34480723323?pr=53716) in relation to this.

**Review by barryhughes — APPROVED** [review]
Looks great, tests well!

> Verify that the logo next to the "Download the WooCommerce App" text is the new one.

✅ 

> Go back to normal display (click the "Toggle device toolbar" icon again) and click the "Get the free WooCommerce mobile app" link in the "What to do next" box. Verify that you see the new logo in the top left corner of the modal window that opens.

✅

> Deactivate the new branding, repeat the steps, and verify that you see the old logos again.

✅

I assume the `pnpm build` issues I noted are local, since they aren't surfacing in CI. Leaving unmerged in case you want to look at the lint issues, however. *Edit: I see you updated already to clear those.*

**Pull request merged**